internal/worker/caasapplicationprovisioner/ops.go

```diff
@@ -708,15 +713,20 @@ func ensureScale(appName string, app caas.Application, appLife life.Value,
 	if err != nil {
 		return err
 	}
+
 	if ps.ScaleTarget >= len(units) {
 		logger.Infof("scaling application %q to desired scale %d", appName, ps.ScaleTarget)
+		if err = app.EnsurePVC(info.Filesystems, info.FilesystemUnitAttachments); err != nil {
```

Comment:
If there's an error scaling below, we would normally look to roll back and changes made here. The `EnsurePVC()` method would need to return a cleanup function.